src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/FinalLocalVariable/case1/context/defaultContextConfig.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+          "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="FinalLocalVariable">
+            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
+        </module>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="context" />
+            <property name="neverSuppressedChecks" value="FinalLocalVariableCheck" />
```

Comment:
This check should not be a neverSuppresedCheck.  It should be added to `checkNameForContextStrategyByTokenOrParent` instead

Reply:
@baratali , yes, one problems is that now current version is like this one:
```java
if (checkNamesForContextStrategyByTokenOrParentSet.contains(checkShortName)) {
                eventAst = getEventAst(event).getParent();
            }
```
we just get violation nodes' parent node, but there is a case that in https://github.com/checkstyle/patch-filters/pull/167#issue-457640353, `IDENT -> x [4:23]` and `IDENT -> y [4:30]`'s parent node is `PARAMETERS [4:19]`, still not `METHOD_DEF [4:4]`. So one solution is use `neverSuppresedCheck` property, and maybe another solution is to add condition, if parent node is PARAMETERS, then get its parent node again. please give me your idea

Reply:
@HuGanghui , hm, I think this method should go up to METHOD_DEF/CTOR_DEF/etc node instead of just getting a parent node. What do you think?
FinalLocalVariable usually produces a lot of violations in a project, so I think a user will not include it to neverSuppressedChecks.

Reply:
@baratali , if we `neverSuppressedChecks` , not all violations in this project will be reported, just those changed/added in the latest commit. And if we choose to give `FinalLocalVariable` special treatment,  then it will have a hardcode in code, is it ok?

Reply:
@HuGanghui , 
> not all violations in this project will be reported, just those changed/added

yeah, I mean that violations will be printed for all variables in a changed file.

> And if we choose to give FinalLocalVariable special treatment, then it will have a hardcode in code, is it ok?

No, we don't need to hardcode any checks. This logic will work for all checks included to `checkNameForContextStrategyByTokenOrParent`. I suggest to specify a collection of node types which should be used for context strategy (METHOD_DEF, CTOR_DEF, VARIABLE_DEF, etc) and if a check is included to 
 `checkNameForContextStrategyByTokenOrParent` then we go up by AST until we find a node whose type is from context node types. So it can be its parent, or parent's parent, or further.
Collection of context node types can be hardcoded because even thirdparty Checks always use existing types from TokenTypes.java.
(Then we should rename `checkNameForContextStrategyByTokenOrParent` probably)

Reply:
yes, I agree with your idea, I will try it.

Reply:
done